lib/board.rb

```diff
@@ -0,0 +1,94 @@
+require 'pry'
+
+class Board
+    attr_accessor :cells, :token
+    
+    def initialize
+        
+        @cells = [" "," "," "," "," "," "," "," "," "]
+
+    end
+ 
+    def reset!
+        cells.clear
+        self.cells = [" "," "," "," "," "," "," "," "," "]
+        
+        
+    end
+    
+    
+    def display
+
+     puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+     puts "-----------"
+     puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+     puts "-----------"
+     puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+   end
+
+#    def display_preturn
+#      puts " 1 | 2 | 3 "
+#      puts "-----------"
+#      puts " 4 | 5 | 6 "
+#      puts "-----------"
+#      puts " 7 | 8 | 9 "
+    
+#     end
+    
+    def position(input)
+
+    "#{@cells[input.to_i-1]}"
+    end
+
+
+
+    def full?
+      @cells.all? do |space|
+        space == "X" || space == "O"
+         end
+    end
+
+    def turn_count
+
+        x_count = cells.count "X"  
+        o_count = cells.count "O"
+        x_count + o_count
+    
+    
+   end
+
+   def taken?(index_number)
+        index = index_number.to_i - 1
+             if @cells[index] == " " || @cells[index] == "" || @cells[index] == nil
+            false
+             elsif @cells[index]== "X" || "O"
+          true
+          
+      end
+     end
+
+     def valid_move?(index_number)
+               
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
input.to_i.between?(1,9) && !taken(input)
```